src/newtondynamics_physics/SysNewton.cpp

```diff
@@ -51,139 +56,177 @@ using osp::active::ACompTransformMutable;
 using osp::Matrix4;
 using osp::Vector3;
 
-// Callback called for every Rigid Body (even static ones) on NewtonUpdate
+// Callback called for dynamic rigid bodies for applying force and torque
 void cb_force_torque(const NewtonBody* pBody, dFloat timestep, int threadIndex)
 {
-    // Get Scene from Newton World
-    ActiveScene* scene = static_cast<ActiveScene*>(
-                NewtonWorldGetUserData(NewtonBodyGetWorld(pBody)));
+    // Get context from Newton World
+    ACtxNwtWorld* pWorldCtx = static_cast<ACtxNwtWorld*>(
```

Comment:
could do "auto * const" without losing clarity here, if you wanted to.

Reply:
it also looks like you only use this in a read-only context, so "auto const*const"